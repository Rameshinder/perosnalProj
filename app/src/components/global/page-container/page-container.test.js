import React from 'react'
import { shallow, mount } from 'enzyme'

import PageContainer from './index'

jest.mock('@anz/icon/filled/generic/print', () => 'PrintIcon')
jest.mock('@anz/button', () => 'Button')

describe('PageContainer', () => {
  const footerProps = {
    showPrimaryButton: false,
    showSecondaryButton: true,
    onPrimaryButtonClick: jest.fn,
    onSecondaryButtonClick: jest.fn,
    primaryButtonText: 'Apply',
    secondaryButtonText: 'Cancel'
  }

  const app = shallow(<PageContainer />)

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })

  it('Print should render without crashing', () => {
    shallow(<PageContainer withActionBar={false} />)
  })

  it('PageContainer should render with action bar', () => {
    mount(<PageContainer withActionBar footerProps={footerProps} />)
  })
})
