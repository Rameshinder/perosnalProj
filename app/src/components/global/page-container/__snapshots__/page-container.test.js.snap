// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
